### PR TITLE
Fix MinioServer health monitor pulse

### PR DESCRIPTION
**Pull request by furkansahin**
**Add log message to MinioServer health checks when exception thrown**
At the moment, we blindly return false and assume the server is down for
MinioServer healthchecks when there is an exception. However, this may
not mean the server is down. Especially considering we have made some
changes to the client, enabled SSL on the Minio side recently, there is
not a clear way to understand what is going on if we simply swallow the
error. This commit adds a clog message for that scenario.

**Fix MinioServer pulse check** 
Health monitor pulse check is still using ip4 address of the
MinioServer. With the current ssl related change, this is updated and we
must use the server_url.

**Refactor MinioServer health check client initialization** 
We had almost the same code in two different places. We move it into a
method in the module and make use of that in the nexus.

### CODE REVIEW

File: prog/minio/minio_server_nexus.rb [hunk 1]
```diff
@@ -171,16 +171,10 @@ def cluster
...
-      secret_key: minio_server.cluster.admin_password,
-      ssl_ca_file_data: minio_server.cluster.root_certs
-    )
-
-    server_data = JSON.parse(client.admin_info.body)["servers"].find { _1["endpoint"] == minio_server.endpoint }
+    server_data = JSON.parse(minio_server.client.admin_info.body)["servers"].find { _1["endpoint"] == minio_server.endpoint }
     server_data["state"] == "online" && server_data["drives"].all? { _1["state"] == "ok" }
-  rescue
+  rescue => e
+    Clog.emit("Minio server is down") { {minio_server: {minio_server_ubid: minio_server.ubid, error: e.message}} }
```
Comment (enescakir) [inline prog/minio/minio_server_nexus.rb:177]: https://github.com/ubicloud/ubicloud/pull/632

IMHO, `exception` isn't a property of `minio_server`. 
Here's a good example for Clog philosophy: https://github.com/ubicloud/ubicloud/blob/756b791f36d970455200697cd9fe78acbd787e49/bin/monitor#L25 
```
 { well_known_metadata_or_representation_name: { ... },
   going_concern: { a_field: "data", b_field: "more data" } }
```

If we think `minio_server` as `well_known_metadata_or_representation_name`
```suggestion
    Clog.emit("Minio server is down") { {minio_server: {ubid: minio_server.ubid}, unavailable_minio_server: {exception: Util.exception_to_hash(ex)}} }
```

If we think it as `going_concern`: `minio_server_down` or `unavailable_minio_server` etc.
```suggestion
    Clog.emit("Minio server is down") { {minio_server_down: {minio_server_ubid: minio_server.ubid, exception: Util.exception_to_hash(ex)}} }
```

Comment (byucesoy) [reply prog/minio/minio_server_nexus.rb:177]: I think you don't even need minio_server_ prefix in your second example.

```
Clog.emit("Minio server is down") { {minio_server_down: {ubid: minio_server.ubid, exception: Util.exception_to_hash(ex)}} }
```

**Review by byucesoy — APPROVED** [review]
Apart from the Enes' comment, LGTM.

File: prog/minio/minio_server_nexus.rb (hunk 1 above)
Comment (furkansahin) [reply prog/minio/minio_server_nexus.rb]: Thanks guys, updated.

**Review by enescakir — APPROVED** [review]

**Pull request merged**